ingest/ingest_files.py

```diff
@@ -0,0 +1,126 @@
+        file_connections = {
+            # Remove BOM with encoding='utf-8-sig'
+            'text/csv': self.open_csv(open_file),
+            'text/plain': open_file,
+            'text/tab-separated-values': self.open_tsv(open_file),
+        }
+        # Check file type
+        file_type = self.get_file_type(file_path)[0]
+        # See if file type is allowed
+        if file_type in self.allowed_file_types:
+            # Return file object and type
+            return file_type, file_connections.get(file_type)
+        else:
+            raise ValueError('Unsupported file format')
+
+    # Inherited function
+    def extract(self):
+        """ Calls extract function for txt, csv, or tsv formmatted files to
+            retrieve all contents from file.
+        """
+
+        file_type_extract_fns = {
+            'text/csv': self.extract_csv_or_tsv,
+            'text/plain': self.extract_txt,
+            'text/tab-separated-values': self.extract_csv_or_tsv,
+        }
+        return file_type_extract_fns.get(self.file_type)()
+
+    def split_line(self, line):
+        """Splits lines on spaces, tabs, or commas"""
+        
+        if self.is_MTX:
+            return re.findall(r'[^,\t]+', line)
+        else:
+            return re.findall(r'[^,\s\t]+', line)
+
+
+    def get_file_type(self, file_path):
+        """Returns file type"""
+        return mimetypes.guess_type(file_path)
+
+    def open_csv(self, opened_file_object):
+        """Opens csv files"""
+        csv.register_dialect('csvDialect',
+                             delimiter=',',
+                             quoting=csv.QUOTE_ALL,
+                             skipinitialspace=True)
+        return csv.reader(opened_file_object, dialect='csvDialect')
+
+    def open_tsv(self, opened_file_object):
+        """Opens tsv files"""
+        csv.register_dialect('tsvDialect',
+                             delimiter='\t',
+                             quoting=csv.QUOTE_ALL,
+                             skipinitialspace=True)
+        return csv.reader(opened_file_object, dialect='tsvDialect')
+
+    def extract_csv_or_tsv(self):
+        """Extracts all rows from a cvs or tsv file"""
+        while(True):
+            try:
+                row = next(self.file)
+                return row
+            except StopIteration:
+                break
+
+    def extract_txt(self):
+        """Extracts all lines from txt files
+
+        Returns:
+                next_row_revised : List[str]
+                    A single row from a txt file .
+        """
+        while True:
+            next_row = self.file.readline()
+
+            if not next_row:
+                break
+            self.amount_of_lines += 1
+            # Create array with no new line, commas, or tab characters
+            next_row_revised = self.split_line(next_row.replace('\n', ''))
+            return next_row_revised
+
+    def get_next_line(self, *, increase_line_count=True, split_line=True):
```

Comment:
This is the first time I recall seeing a [bare asterisk in function arguments](https://stackoverflow.com/a/14298976/10564415).  Neat use of this new Python 3 feature to force the caller to use named arguments!